gravitee-am-repository/gravitee-am-repository-mongodb/src/main/java/io/gravitee/am/repository/mongodb/management/AbstractUserRepository.java

```diff
@@ -532,4 +538,32 @@ private List<CertificateMongo> toMongoCertificates(List<Certificate> modelCertif
                     return mongoCertificate;
                 }).collect(Collectors.toList());
     }
+
+    private void createOrUpdateIndex() {
+        if (ensureIndexOnStart) {
+            getDeletableIndex()
+                    .doOnComplete(() -> {
+                        try {
+                            super.createIndex(usersCollection, new Document(FIELD_REFERENCE_TYPE, 1).append(FIELD_REFERENCE_ID, 1).append(FIELD_USERNAME, 1).append(FIELD_SOURCE, 1),
+                                    new IndexOptions().name(INDEX_REFERENCE_TYPE_REFERENCE_ID_USERNAME_SOURCE_NAME).unique(true));
+                        } catch (Exception e) {
+                            logger.error("An error has occurred while creating index {} with unique constraints", INDEX_REFERENCE_TYPE_REFERENCE_ID_USERNAME_SOURCE, e);
+                        }
+                    }).doOnError(e -> logger.error("An error has occurred while deleting index {}", INDEX_REFERENCE_TYPE_REFERENCE_ID_USERNAME_SOURCE, e))
+                    .subscribe();
+        }
+    }
+
+    private Completable getDeletableIndex() {
+        return Observable.fromPublisher(usersCollection.listIndexes())
+                .map(document -> document.getString("name"))
+                .flatMapCompletable(indexName -> {
+                    if (indexName.equals(AbstractUserRepository.INDEX_REFERENCE_TYPE_REFERENCE_ID_USERNAME_SOURCE) ||
+                            indexName.equals(INDEX_REFERENCE_TYPE_REFERENCE_ID_USERNAME_SOURCE_NAME)) {
```

Comment:
You should test only the `INDEX_REFERENCE_TYPE_REFERENCE_ID_USERNAME_SOURCE` otherwise by testing also the `INDEX_REFERENCE_TYPE_REFERENCE_ID_USERNAME_SOURCE_NAME`, the index will be deleted and created after each restart